src/prop/sat_proof_manager.cpp

```diff
@@ -0,0 +1,729 @@
+/*********************                                                        */
+/*! \file sat_proof_manager.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Haniel Barbosa
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Implementation of the proof manager for Minisat
+ **/
+
+#include "prop/sat_proof_manager.h"
+
+#include "expr/proof_node_algorithm.h"
+#include "options/smt_options.h"
+#include "prop/cnf_stream.h"
+#include "prop/minisat/minisat.h"
+#include "prop/theory_proxy.h"
+#include "theory/theory_proof_step_buffer.h"
+
+namespace CVC4 {
+namespace prop {
+
+SatProofManager::SatProofManager(Minisat::Solver* solver,
+                                 CVC4::prop::TheoryProxy* proxy,
+                                 context::UserContext* userContext,
+                                 ProofNodeManager* pnm)
+    : d_solver(solver),
+      d_proxy(proxy),
+      d_pnm(pnm),
+      d_resChains(pnm, true, userContext),
+      d_resChainPg(userContext, pnm),
+      d_assumptions(userContext),
+      d_conflictLit(undefSatVariable)
+{
+  d_false = NodeManager::currentNM()->mkConst(false);
+}
+
+void SatProofManager::printClause(const Minisat::Clause& clause)
+{
+  for (unsigned i = 0, size = clause.size(); i < size; ++i)
+  {
+    SatLiteral satLit = MinisatSatSolver::toSatLiteral(clause[i]);
+    Trace("sat-proof") << satLit << " ";
+  }
+}
+
+Node SatProofManager::getClauseNode(SatLiteral satLit)
+{
+  Assert(d_proxy->getCnfStream()->getNodeCache().find(satLit)
```

Comment:
Can you cache the CNF stream of d_proxy as a data member of this class?  It seems it is used very heavily.

Reply:
Actually, I'm curious if you can just pass the CNF stream to the constructor of this class instead of d_proxy.